common/src/primitives/amount.rs

```diff
@@ -156,6 +156,19 @@ impl std::ops::Shr<u32> for Amount {
     }
 }
 
+impl std::iter::Sum<Amount> for Option<Amount> {
+    fn sum<I>(mut iter: I) -> Self
+    where
+        I: Iterator<Item = Amount>,
+    {
+        iter.try_fold(Amount::new(0), std::ops::Add::add)
+    }
+}
+
+pub fn random(range: std::ops::RangeInclusive<Amount>) -> Amount {
+    rand::thread_rng().gen_range(range.start().val..=range.end().val).into()
+}
```

Comment:
Minor style comment: Could this be a member function of Amount (with no `self` argument)?

Reply:
Done, in my new branch, so you won't see it here... Not sure if to override this PR with the new branch and set it as a draft. Will consult with Sam.